### ISSUE TITLE
Way to export/save intermediate logit outputs

### ISSUE THREAD

**Issue by EomSooHwan** [NVIDIA/NeMo#7432]
Hello, I am currently training the Conformer-CTC model for ASR.

I am interested in how logit prediction changes throughout training, and I am wondering if there is any way to modify the code so that my model saves logits and targets data as a pickle file for the end of each validation step. I am aware that I can append each (logits, targets) in `validation_step_outputs` and then save them in `on_validation_epoch_end()`. However, I am a bit scared to overwrite the function in `ctc_bpe_models.py`. It would be great if there is a safer way to handle such a situation.

**Comment by titu1994** [comment]
You don't need to modify that class, you can simply subclass it in your own python file, modify the code needed in subclass (the two validation methods) and probably also validation step, and then simply do subclass.restore_from(CTC model) - it will then use your subclass to perform training and so store data in validation step c

One thing to be careful off is writing files under ddp should be done only by rank 0 otherwise you'll corrupt files across multiple DDP ranks.
